pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -0,0 +1,148 @@
+			masqueradeConfigurator *MasqueradePodNetworkConfigurator
+			iface                  *v1.Interface
+			podLink                *netlink.GenericLink
+			vmi                    *v1.VirtualMachineInstance
+			ipv4GwIP               *netlink.Addr
+			ipv6GwIP               *netlink.Addr
+			ipv4VmIP               *netlink.Addr
+			ipv6VmIP               *netlink.Addr
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIMasqueradeInterface("default", "vm1")
+
+			masqueradeConfigurator = NewMasqueradePodNetworkConfigurator(vmi, iface, bridgeIfaceName, nil, ipv4CidrToConfigure, ipv6CidrToConfigure, cacheFactory, launcherPID, handler)
+			ipv4GwIP, _ = netlink.ParseAddr(ipv4GwAddr)
+			ipv6GwIP, _ = netlink.ParseAddr(ipv6GwAddr)
+			ipv4VmIP, _ = netlink.ParseAddr(vmIPv4Addr)
+			ipv6VmIP, _ = netlink.ParseAddr(vmIPv6Addr)
+		})
+
+		When("the pod link is defined", func() {
+			BeforeEach(func() {
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+				handler.EXPECT().LinkByName(ifaceName).Return(podLink, nil)
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, fmt.Errorf("failed to compute addresses"))
+				})
+
+				It("should fail to read the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+				})
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, nil)
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMIpv6CIDR).Return(ipv6GwAddr, vmIPv6Addr, fmt.Errorf("failed to compute addresses"))
+					handler.EXPECT().ParseAddr(ipv4GwAddr).Return(ipv4GwIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv4Addr).Return(ipv4VmIP, nil)
+					handler.EXPECT().IsIpv6Enabled(ifaceName).Return(true, nil)
+
+				})
+
+				It("should fail to read the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+				})
+			})
+
+			When("the pod interface only has an IPv4 address", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, nil)
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMIpv6CIDR).Return(ipv6GwAddr, vmIPv6Addr, nil).Times(0)
+					handler.EXPECT().ParseAddr(ipv4GwAddr).Return(ipv4GwIP, nil)
+					handler.EXPECT().ParseAddr(ipv6GwAddr).Return(ipv6GwIP, nil).Times(0)
+					handler.EXPECT().ParseAddr(vmIPv4Addr).Return(ipv4VmIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv6Addr).Return(ipv6VmIP, nil).Times(0)
+					handler.EXPECT().IsIpv6Enabled(ifaceName).Return(false, nil)
+				})
+
+				It("should succeed reading the pod link info", func() {
```

Comment:
"should succeed reading the pod link IPv4 info"

Also add checks for gateway and pod address for IPv4 also check that IPv6 is not set.

Reply:
? this context features IPv6 addresses. I don't follow.